src/js/angular/rest/ttyg.rest.service.js

```diff
@@ -65,14 +65,14 @@ function TTYGRestService($http) {
 
     /**
      * Calls the REST API to ask a question.
-     * @param {*} data
-     * @return {*}
+     * @param {ChatQuestion} chatQuestion
+     * @return {Promise<ChatMessageModel>}
```

Comment:
This is not correct return type. Technically the shape of the response might be similar, but should not be taken for ganted and we should not entangle with its format, so better remove the type or define a separate chat message response model. 
The other thing is that the reponse conversion from plain JSON to the ChatMessageMode is actually not performed here, but in the other service which is another reason for removing this type from here because it's misguiding.

Reply:
Reverted